File: src/main/java/com/mpe85/grampa/input/StringInputBuffer.java
package com.mpe85.grampa.input;

public class StringInputBuffer extends AbstractCharSequenceInputBuffer {
	
	public StringInputBuffer(final String string) {
		super(string);
		this.string = string;
	}
	
	@Override
	public int getCodePoint(final int index) {
		return string.codePointAt(index);
	}
	
	
	private final String string;
	
}

File: src/main/kotlin/com/github/mpe85/grampa/input/StringInputBuffer.kt
package com.github.mpe85.grampa.input

/**
 * An [InputBuffer] implementation using a [String].
 *
 * @property[string] A string backing the input buffer
 * @author mpe85
 */
public class StringInputBuffer(private val string: String) : CharSequenceInputBuffer(string) {

    override fun getCodePoint(index: Int): Int = string.codePointAt(index)
}
